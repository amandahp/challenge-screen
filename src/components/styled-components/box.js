import styled from 'styled-components'

export const Box = styled.div`
  ${() => `
  display: flex;
  margin-top: 70px;
  align-items: center;
  flex-direction: column;
  h6{
    font-family: 'Lato', sans-serif;
    font-size:75%;
    letter-spacing: 1.3px;
    margin-top: 15px;
  }
  `}
`
