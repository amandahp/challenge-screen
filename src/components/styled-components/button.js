import styled from 'styled-components'

export const Button = styled.button`
  ${({ theme }) => `
    background: rgb(97,0,135);
    background: linear-gradient(90deg, rgba(97,0,135,1) 33%, rgba(77,0,140,1) 73%, rgba(56,0,144,1) 100%);
    color: ${theme.color?.light};
    font-family: 'Lato', sans-serif;
    width: 46%;
    height: 60px;
    font-size: 1.2rem;
    font-weight:bold;
    margin-top: -20px;
    border-radius: 30px;
    cursor: pointer;
    border:1px solid ${theme.color?.darkBlue};
    &:hover{
      background-color: ${theme.color?.light};
      color: ${theme.color?.darkBlue};
    };
    &:focus{
      background-color: ${theme.color?.light};
      color: ${theme.color?.darkBlue};
      outline: none;
    };
    @media (max-width:500px) {
      font-size: 0.8rem;
      width:60%;
    }
  `}
`
