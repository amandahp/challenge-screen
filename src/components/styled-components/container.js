import styled from 'styled-components'

export const Container = styled.div`
  ${() => `
  display: grid;
  grid-template-columns: repeat(5, 1fr);
  grid-gap: 1rem;
  grid-auto-rows: repeat(5, minmax(100px, auto))
  margin-top: -90;

  @media(max-width: 800px) {
    grid-template-columns: repeat(1, 1fr);
    grid-gap: 1rem;
  }
  `}
`
