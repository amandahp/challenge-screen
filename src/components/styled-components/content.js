import styled from 'styled-components'

export const Content = styled.div`
  ${({ theme }) => `
    margin: auto;
    width: 80%;
    height: 75%;
    border-radius: 20px;
    position: relative;
    ${theme.bodyLargeLoose}
    background: ${theme.backgroundColor?.primary};
    padding: 170px;
    align-content: center;
    @media(max-width: 986px){
      height: auto;
      width: 80%;
      padding: 100px;
      display:flex;
      flex-direction:column;
    }
  `}
`
