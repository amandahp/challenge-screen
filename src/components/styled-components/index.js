export { Wrapper } from './wrapper'
export { Content } from './content'
export { Span } from './span'
export { Container } from './container'
export { Button } from './button'
export { Box } from './box'
export { NewBox } from './newBox'
export { Text } from './text'
