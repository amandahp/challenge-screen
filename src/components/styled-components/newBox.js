import styled from 'styled-components'

export const NewBox = styled.div`
  ${({ theme }) => `
  margin-bottom: 25px;
  align-items: center;
  flex-direction: column;
  h5{
      margin-top: 40px;
      font-family: 'Lato', sans-serif;
      font-size:120%;
      color: ${theme.color?.purple}
  }

  `}
`
