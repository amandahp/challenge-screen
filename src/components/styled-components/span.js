import styled from 'styled-components'

export const Span = styled.span`
  ${({ theme }) => `
    margin: 0 auto;
    border-radius: 10px;
    width: 45%;
    height: 10%;
    position: absolute;
    top: -4%;
    right: 29%;
    text-align: center;
    ${theme.bodyLargeLoose}
    color: ${theme.color?.light};
    background: ${theme.color?.darkBlue};
    display: flex;
    align-content: center;
    align-items: center;
    justify-content: center;
    h4 {
      font-size:1.5em;
      font-family: 'Lato', sans-serif;
    }
    @media(max-width:976px) {
      top:0;
    }
  `}
`
