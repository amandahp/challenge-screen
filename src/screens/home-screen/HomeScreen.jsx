import React, { useState, useEffect, useCallback } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { fetchData } from 'store/ducks/home/actions'

import { Loader } from 'components/loader/loader'
import * as C from 'components'
import * as Styled from 'components/styled-components'

export const HomeScreen = () => {
  const [loanValue, setLoan] = useState('')
  const [rawLoanValue, setRawLoanValue] = useState('')
  const [installmentValue, setInstallmentValue] = useState('')
  const [selectedInstallment, setSelectedInstallment] = useState('')

  const dispatch = useDispatch()

  const data = useSelector((state) => state.homeReducer.data)
  const loading = useSelector((state) => state.homeReducer.loading)
  const error = useSelector((state) => state.homeReducer.error)

  const { valoresEmprestimo, parcelas, valoresEmprestimeBruto, prazos } = data

  const setInitialValues = useCallback(() => {
    setLoan(valoresEmprestimo[2])
    setRawLoanValue(valoresEmprestimeBruto[2])
    setInstallmentValue(parcelas[2].reverse())
  }, [valoresEmprestimo, valoresEmprestimeBruto, parcelas])

  useEffect(() => {
    dispatch(fetchData())
  }, [dispatch])

  useEffect(() => {
    if (data) {
      setInitialValues()
    }
  }, [data, setInitialValues])

  const handleChangeSlider = (value) => {
    switch (value) {
      case 0:
        setLoan(valoresEmprestimo[0])
        setRawLoanValue(valoresEmprestimeBruto[0])
        setInstallmentValue(parcelas[0].reverse())
        break
      case 25:
        setLoan(valoresEmprestimo[1])
        setRawLoanValue(valoresEmprestimeBruto[1])
        setInstallmentValue(parcelas[1].reverse())
        break
      case 50:
        setLoan(valoresEmprestimo[2])
        setRawLoanValue(valoresEmprestimeBruto[2])
        setInstallmentValue(parcelas[2].reverse())
        break
      case 75:
        setLoan(valoresEmprestimo[3])
        setRawLoanValue(valoresEmprestimeBruto[3])
        setInstallmentValue(parcelas[3].reverse())
        break
      case 100:
        setLoan(valoresEmprestimo[4])
        setRawLoanValue(valoresEmprestimeBruto[4])
        setInstallmentValue(parcelas[4].reverse())
        break
    }
  }

  const handleSelectInstallment = (installment) => {
    setSelectedInstallment(installment)
  }

  const handleSubmit = () => {
    console.log(selectedInstallment, loanValue)
  }

  return (
    <Styled.Wrapper>
      <Styled.Content>
        <Styled.Span>
          <h4>Valor Solicitado</h4>
        </Styled.Span>
        {valoresEmprestimo && !loading ? (
          <>
            <Styled.NewBox>
              <C.BoxLoan valueLoan={loanValue} valueRaw={rawLoanValue} />
              <C.SliderComponent
                loanOptions={valoresEmprestimo}
                handleChange={handleChangeSlider}
              />
              <Styled.Text>
                <h5>Selecione a quantidade de parcelas</h5>
              </Styled.Text>
            </Styled.NewBox>
            <Styled.Container>
              {prazos.map((prazo, index) => {
                return (
                  <C.ButtonInstallment
                    key={index + 1}
                    handleClick={handleSelectInstallment}
                    valueMonth={prazo}
                    valueInstallment={installmentValue[index]}
                  />
                )
              })}
            </Styled.Container>
            <Styled.Box>
              <Styled.Button onClick={() => handleSubmit()}>
                Gostei, continuar
              </Styled.Button>
              <h6>
                Taxa de 1.09% ao mês. Valor da primeira - Sistema de
                Automatização constante (suas parcelas diminuem com o tempo)
              </h6>
            </Styled.Box>
          </>
        ) : (
          <Styled.Box>
            <Loader />
          </Styled.Box>
        )}
        {error && <p>erro</p>}
      </Styled.Content>
    </Styled.Wrapper>
  )
}
