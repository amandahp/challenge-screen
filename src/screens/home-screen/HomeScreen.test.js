import React from 'react'
import { render, screen } from '@testing-library/react'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import { ThemeProvider } from 'styled-components'
import themePontte from 'theme/theme'

const mockStore = configureStore([])

import '@testing-library/jest-dom/extend-expect'

import { HomeScreen } from './HomeScreen'

describe('<HomeScreen />', () => {
  let store

  beforeEach(() => {
    store = mockStore({
      homeReducer: {
        data: {
          prazos: [6, 9],
          valoresEmprestimo: [70, 85, 90, 100, 120],
          valoresEmprestimeBruto: [78, 95],
          parcelas: [
            [2287.12, 2736.97, 3173.81, 3605.04, 4036.26],
            [1848.68, 2208.55, 2558.02, 2903, 3247.98],
            [1848.68, 2208.55, 2558.02, 2903, 3247.98]
          ]
        }
      }
    })
  })

  it('should render the heading', () => {
    render(
      <Provider store={store}>
        <HomeScreen />{' '}
      </Provider>
    )
    expect(
      screen.getByRole('heading', {
        name: /Selecione a quantidade de parcelas/i
      })
    ).toBeInTheDocument
  })
  it('should render the heading button', () => {
    render(
      <Provider store={store}>
        <HomeScreen />{' '}
      </Provider>
    )

    expect(
      screen.getByRole('button', {
        name: /Gostei, continuar/i
      })
    ).toBeInTheDocument
  })
  it('should render the heading', () => {
    render(
      <Provider store={store}>
        <HomeScreen />{' '}
      </Provider>
    )
    expect(
      screen.getByRole('heading', {
        name: /Valor Solicitado/i
      })
    ).toBeInTheDocument
  })

  it('should render matching the snapshot', () => {
    const { container } = render(
      <Provider store={store}>
        <ThemeProvider theme={themePontte}>
          <HomeScreen />
        </ThemeProvider>
      </Provider>
    )
    expect(container).toMatchSnapshot()
  })
})
