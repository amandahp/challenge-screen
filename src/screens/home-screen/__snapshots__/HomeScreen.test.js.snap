// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> should render matching the snapshot 1`] = `
.c4 {
  margin: 0 auto;
  width: 45%;
  height: 10%;
  position: absolute;
  right: 29%;
  top: 15%;
  text-align: center;
  color: #03033A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-size: 1.5em;
  font-family: 'Lato',sans-serif;
  line-height: 1.5;
}

.c4 button {
  color: #03033A;
  border: 1px solid #03033A;
  background-color: #FFFFFF;
  width: 30px;
  height: 30px;
  border-radius: 100%;
  cursor: pointer;
}

.c4 button:hover {
  background-color: #03033A;
  color: #FFFFFF;
}

.c4 button:focus {
  outline: none;
}

.c5 {
  width: 500px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c8 {
  background-color: #FFFFFF;
  font-family: 'Lato',sans-serif;
  width: 100%;
  height: 185%;
  border-radius: 10px;
  cursor: pointer;
  border: 1px solid #03033A;
}

.c8 h3 {
  font-weight: bold;
  font-size: 1em;
}

.c8:hover {
  background-color: #03033A;
  color: #FFFFFF;
}

.c8:focus {
  background-color: #03033A;
  color: #FFFFFF;
  outline: none;
}

.c0 {
  width: 100%;
  height: 100vh;
  background: rgb(97,0,135);
  background: linear-gradient(90deg,rgba(97,0,135,1) 33%,rgba(77,0,140,1) 73%,rgba(56,0,144,1) 100%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center flex-direction:column;
  -webkit-box-align: center flex-direction:column;
  -ms-flex-align: center flex-direction:column;
  align-items: center flex-direction:column;
}

.c1 {
  margin: auto;
  width: 80%;
  height: 75%;
  border-radius: 20px;
  position: relative;
  text-transform: none;
  font-style: normal;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  background: #FFFFFF;
  padding: 170px;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c2 {
  margin: 0 auto;
  border-radius: 10px;
  width: 45%;
  height: 10%;
  position: absolute;
  top: -4%;
  right: 29%;
  text-align: center;
  text-transform: none;
  font-style: normal;
  -webkit-letter-spacing: 0rem;
  -moz-letter-spacing: 0rem;
  -ms-letter-spacing: 0rem;
  letter-spacing: 0rem;
  color: #FFFFFF;
  background: #03033A;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 h4 {
  font-size: 1.5em;
  font-family: 'Lato',sans-serif;
}

.c7 {
  display: grid;
  grid-template-columns: repeat(5,1fr);
  grid-gap: 1rem;
  grid-auto-rows: repeat(5,minmax(100px,auto)) margin-top:-90;
}

.c10 {
  background: rgb(97,0,135);
  background: linear-gradient(90deg,rgba(97,0,135,1) 33%,rgba(77,0,140,1) 73%,rgba(56,0,144,1) 100%);
  color: #FFFFFF;
  font-family: 'Lato',sans-serif;
  width: 46%;
  height: 60px;
  font-size: 1.2rem;
  font-weight: bold;
  margin-top: -20px;
  border-radius: 30px;
  cursor: pointer;
  border: 1px solid #03033A;
}

.c10:hover {
  background-color: #FFFFFF;
  color: #03033A;
}

.c10:focus {
  background-color: #FFFFFF;
  color: #03033A;
  outline: none;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-top: 70px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c9 h6 {
  font-family: 'Lato',sans-serif;
  font-size: 75%;
  -webkit-letter-spacing: 1.3px;
  -moz-letter-spacing: 1.3px;
  -ms-letter-spacing: 1.3px;
  letter-spacing: 1.3px;
  margin-top: 15px;
}

.c3 {
  margin-bottom: 25px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 h5 {
  margin-top: 40px;
  font-family: 'Lato',sans-serif;
  font-size: 120%;
  color: #850083;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (max-width:986px) {
  .c4 {
    position: relative;
    right: 0%;
    margin-bottom: 10px;
  }
}

@media (max-width:500px) {
  .c4 {
    font-size: 18px;
  }
}

@media (max-width:800px) {

}

@media (max-width:800px) {
  .c0 {
    height: auto;
    width: 100%;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-box-pack: center;
    -webkit-justify-content: center;
    -ms-flex-pack: center;
    justify-content: center;
    -webkit-align-items: center flex-direction:column;
    -webkit-box-align: center flex-direction:column;
    -ms-flex-align: center flex-direction:column;
    align-items: center flex-direction:column;
  }
}

@media (max-width:986px) {
  .c1 {
    height: auto;
    width: 80%;
    padding: 100px;
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media (max-width:976px) {
  .c2 {
    top: 0;
  }
}

@media (max-width:800px) {
  .c7 {
    grid-template-columns: repeat(1,1fr);
    grid-gap: 1rem;
  }
}

@media (max-width:500px) {
  .c10 {
    font-size: 0.8rem;
    width: 60%;
  }
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <span
        class="c2"
      >
        <h4>
          Valor Solicitado
        </h4>
      </span>
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <span>
            <h2>
              R$ 90,00
            </h2>
            <div
              class="c5"
            >
              <h3>
                Valor bruto:
                
              </h3>
              <button>
                ?
              </button>
            </div>
          </span>
        </div>
        <div>
          <div
            class="rc-slider rc-slider-with-marks"
          >
            <div
              class="rc-slider-rail"
              style="background-color: rgb(146, 47, 145);"
            />
            <div
              class="rc-slider-track"
              style="background-color: rgb(164, 203, 80); left: 0%; width: 50%;"
            />
            <div
              class="rc-slider-step"
            >
              <span
                class="rc-slider-dot rc-slider-dot-active"
                style="border-color: white; left: 0%;"
              />
              <span
                class="rc-slider-dot rc-slider-dot-active"
                style="border-color: white; left: 25%;"
              />
              <span
                class="rc-slider-dot rc-slider-dot-active"
                style="border-color: white; left: 50%;"
              />
              <span
                class="rc-slider-dot"
                style="border-color: white; left: 75%;"
              />
              <span
                class="rc-slider-dot"
                style="border-color: white; left: 100%;"
              />
            </div>
            <div
              aria-disabled="false"
              aria-valuemax="100"
              aria-valuemin="0"
              aria-valuenow="50"
              class="rc-slider-handle"
              role="slider"
              style="border-color: #03033a; background-color: rgb(3, 3, 58); left: 50%; transform: translateX(-50%);"
              tabindex="0"
            />
            <div
              class="rc-slider-mark"
            >
              <span
                class="rc-slider-mark-text rc-slider-mark-text-active"
                style="transform: translateX(-50%); left: 0%; color: rgb(16, 16, 68); font-weight: bold;"
              >
                R$ 70,00
              </span>
              <span
                class="rc-slider-mark-text"
                style="transform: translateX(-50%); left: 100%; color: rgb(16, 16, 68); font-weight: bold;"
              >
                R$ 120,00
              </span>
            </div>
          </div>
        </div>
        <div
          class="c6"
        >
          <h5>
            Selecione a quantidade de parcelas
          </h5>
        </div>
      </div>
      <div
        class="c7"
      >
        <button
          class="c8"
        >
          <h4>
            6
             meses
          </h4>
          <h2>
             
            R$ 3.247,98
          </h2>
        </button>
        <button
          class="c8"
        >
          <h4>
            9
             meses
          </h4>
          <h2>
             
            R$ 2.903,00
          </h2>
        </button>
      </div>
      <div
        class="c9"
      >
        <button
          class="c10"
        >
          Gostei, continuar
        </button>
        <h6>
          Taxa de 1.09% ao mês. Valor da primeira - Sistema de Automatização constante (suas parcelas diminuem com o tempo)
        </h6>
      </div>
    </div>
  </div>
</div>
`;
